modules/tensor_mechanics/test/tests/plane_stress/3D_finite_tension_pull.i

```diff
@@ -0,0 +1,126 @@
+[GlobalParams]
+  order = FIRST
+  family = LAGRANGE
+  displacements = 'disp_x disp_y disp_z'
+[]
+
+[Problem]
+  extra_tag_vectors = 'ref'
+[]
+
+[Mesh]
+  [gmg]
+    type = GeneratedMeshGenerator
+    dim = 3
+    nx = 1
+    ny = 1
+    nz = 1
+  []
+[]
+
+[Variables]
+  [disp_x]
+  []
+  [disp_y]
+  []
+  [disp_z]
+  []
+[]
```

Comment:
any reason not to use `add_variables`?

Reply:
I'm not a huge fan of add_variables, and I know others aren't either because it hides an essential part of what the model is doing.